### PR TITLE
Only check for ORC header magic if footer read fails

### PR DESCRIPTION
**Pull request by dain**

### CODE REVIEW

File: presto-orc/src/main/java/com/facebook/presto/orc/OrcReader.java [hunk 1]
```diff
@@ -111,12 +111,18 @@ public OrcReader(OrcDataSource orcDataSource, MetadataReader delegate, DataSize
...
 
-        // make sure this is an ORC file and not an RCFile or something else
-        verifyOrcFooter(orcDataSource, postScriptSize, buffer);
-
         // decode the post script
-        int postScriptOffset = buffer.length - SIZE_OF_BYTE - postScriptSize;
-        PostScript postScript = metadataReader.readPostScript(buffer, postScriptOffset, postScriptSize);
+        PostScript postScript;
+        try {
+            postScript = metadataReader.readPostScript(buffer, buffer.length - SIZE_OF_BYTE - postScriptSize, postScriptSize);
```
Comment (martint) [inline presto-orc/src/main/java/com/facebook/presto/orc/OrcReader.java]: Is there a chance it is not a valid Orc file but this call doesn't fail?

**Comment by electrum** [comment]
Can you add a description to the commit explaining the purpose?

File: presto-orc/src/main/java/com/facebook/presto/orc/OrcReader.java (hunk 1 above)
Comment (dain) [reply presto-orc/src/main/java/com/facebook/presto/orc/OrcReader.java]: No.. it is protobuf with multiple fields

File: presto-orc/src/main/java/com/facebook/presto/orc/OrcReader.java [hunk 2]
```diff
@@ -246,31 +252,15 @@ private static OrcDataSource wrapWithCacheIfTiny(OrcDataSource dataSource, DataS
     }
 
     /**
-     * Verify this is an ORC file to prevent users from trying to read text
-     * files or RC files as ORC files.
+     * Does the file start with the ORC magic bytes.
```
Comment (electrum) [inline presto-orc/src/main/java/com/facebook/presto/orc/OrcReader.java]: Add a question mark at end

**Pull request closed without merging**